lewis/core/devices.py

```diff
@@ -111,21 +111,25 @@ class DeviceBuilder(object):
     def __init__(self, module):
         self._module = module
 
-        self._device_types = list(get_members(self._module, is_device).values())
-
-        if not self._device_types:
-            # syslog.warning('No devices discovered in module {}.'.format(module.__name__))
-            pass
-        else:
-            pass
-            # syslog.info('Discovered devices in module {}: {}'.format(
-            #    module.__name__, ','.join([x.__name__ for x in self._device_types])))
-
         submodules = get_submodules(self._module)
 
+        # Devices can be in __init__.py, device.py, or any devices/*.py
+        self._device_types = list(get_members(self._module, is_device).values())
+        self._device_types += list(get_members(submodules.get('device'), is_device).values())
```

Comment:
You're right, `device.py` can be removed. I left it in because I was thinking of the devices that the IBEX team created (they have a `device.py`). But they should not break anyway because they still have the `__all__` stuff in their `__init__.py`. It just means they have to put their `device.py` into `devices/` if they want to get rid of the `__all__` stuff.

As for putting the additional cases (now only `__init__.py`) into `_discover_devices`... I'm a bit conflicted. I had the same thought as you initially, but decided to leave them separate because I felt that it would be *in*consistent with the other two discovery methods. As they are, they all have a signature like this:

```python
_discover_something(some_module)
```
and return whatever `something`s they found in `some_module`.

If we move `__init__.py` discovery into `_discover_devices`, it will now return whatever it found in `some_module` _plus_ whatever it found in `__init__.py`. If, at some point in the future, we want to also load devices from `some_other_module`, we won't be able to just add one line like this:

```python
self._device_types += self._discover_devices(submodules.get('devices'))
self._device_types += self._discover_devices(submodules.get('some_other_module'))
```
because now we'll have anything found in `__init__.py` returned twice.